shim/src/epoll/epoll_ctl.c

```diff
@@ -38,18 +41,35 @@ static int __do_demi_epoll_ctl_add(int epfd, int fd, struct epoll_event *event)
     for (int i = 0; i < MAX_EVENTS; i++)
     {
         // Found.
-        struct demi_event *ev = epoll_get_event(epfd, i);
+        ev = epoll_get_event(epfd, i);
```

Comment:
Why have you nived `ev` declaration to the outter scope? The lifetime of `ev` does not need to be that long.